tt_eager/tt_dnn/op_library/eltwise_binary/eltwise_binary_op.hpp

```diff
@@ -178,11 +180,11 @@ inline Tensor add(
     std::optional<std::vector<UnaryWithParam>> fused_activations = std::nullopt,
     const MemoryConfig &output_mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG,
     std::optional<const DataType> output_dtype = std::nullopt,
-    bool in_place = false) {
+    bool in_place = false,
```

Comment:
We should remove the in_place flag from eltwise_binary now that we have optional output tensor support instead, as well as removing this version of add function from primary namespace, since there is no reason to have this separate and update any op that was using inplace=True to passing the first input as the output tensor

Reply:
@tt-aho @eyonland Could the in_place flag changes and removing this version of add function be done as a separate issue?   I see a lot of models using this function with in_place=True and the current changes in this PR don't affect them. Implementing optional output tensor in composite and backward ops is dependent on this PR #8671  #8595 #8785

Reply:
I'm ok with that so long as we get it done.  Thanks

Reply:
Sounds good